### DISCUSSION TITLE
Support tab indentation in py-editor

### DISCUSSION THREAD

**Discussion by MarcSkovMadsen** [pyscript/pyscript#1908] [Proposals]
I've just tried using the `py-editor`. When I'm changing the code inside the editor I would expect `tab` to indent my code. But instead I jump to some other html element on the page.

Please support `tab` indentation and `shift+tab` dedentation. And probably more keyboard short cuts.

Personally I would prefer all my normal VS Code keyboard shortcuts to apply 😄

**Comment by JeffersGlass** [comment]
Capturing `tab` breaks default accessibility and screen navigation, unfortunately. I don't think it explicitly violates WCAG, which provides [guidance for instructing the user when an alternate key should be used to navigate out of content](https://www.w3.org/WAI/WCAG22/Techniques/general/G21.html), but I'm not sure it's something we as PyScript should do out of the box... 

Just my opinion; from a technical standpoint it's not too complicated to implement.

**Reply by halessi** [reply]
how might someone implement this? i've spent a bunch of time trying to figure it out (ie, diving into the specifics of what seem to be a CodeMirror implementation) but have not had any luck. interested in binding to ctrl-tab or something, as an option, to avoid breaking accessibility

**Reply by MarcSkovMadsen** [reply]
Here is how CodeMirror handles Tab: https://codemirror.net/examples/tab/. That could be used as inspiration.
